src/main/java/uk/gov/pay/directdebit/mandate/services/gocardless/GoCardlessMandateStateCalculator.java

```diff
@@ -35,8 +39,20 @@
         this.goCardlessEventDao = goCardlessEventDao;
     }
 
-    Optional<DirectDebitStateWithDetails<MandateState>> calculate(GoCardlessMandateIdAndOrganisationId goCardlessMandateIdAndOrganisationId) {
-        return goCardlessEventDao.findLatestApplicableEventForMandate(goCardlessMandateIdAndOrganisationId, GOCARDLESS_ACTIONS_THAT_CHANGE_STATE)
+    Optional<DirectDebitStateWithDetails<MandateState>> calculate(Mandate mandate) {
+        Optional<GoCardlessEvent> latestGoCardlessEvent = mandate.getPaymentProviderMandateId()
```

Comment:
Maybe `latestApplicableGoCardlessEvent` to make it clear it’s not necessarily the absolute latest?